dissect/target/plugins/os/unix/log/atop.py

```diff
@@ -0,0 +1,362 @@
+        count_t vgrow;		/* virtual  growth  (Kb)    	*/
+        count_t rgrow;		/* resident growth  (Kb)     	*/
+        count_t vdata;		/* virtmem data     (Kb)     	*/
+        count_t vstack;		/* virtmem stack    (Kb)     	*/
+        count_t vlibs;		/* virtmem libexec  (Kb)     	*/
+        count_t vswap;		/* swap space used  (Kb)     	*/
+        count_t	vlock;		/* virtual locked   (Kb) 	*/
+        count_t	cfuture[3];	/* reserved for future use	*/
+    } mem;
+
+    /* NETWORK STATISTICS						*/
+    struct net {
+        count_t tcpsnd;		/* number of TCP-packets sent	*/
+        count_t tcpssz;		/* cumulative size packets sent	*/
+        count_t	tcprcv;		/* number of TCP-packets recved	*/
+        count_t tcprsz;		/* cumulative size packets rcvd	*/
+        count_t	udpsnd;		/* number of UDP-packets sent	*/
+        count_t udpssz;		/* cumulative size packets sent	*/
+        count_t	udprcv;		/* number of UDP-packets recved	*/
+        count_t udprsz;		/* cumulative size packets sent	*/
+        count_t	avail1;		/* */
+        count_t	avail2;		/* */
+        count_t	cfuture[4];	/* reserved for future use	*/
+    } net;
+
+    struct gpu {
+        char	state;		// A - active, E - Exit, '\0' - no use
+        char	cfuture[3];	//
+        short	nrgpus;		// number of GPUs for this process
+        int32_t	gpulist;	// bitlist with GPU numbers
+        int	gpubusy;	// gpu busy perc process lifetime      -1 = n/a
+        int	membusy;	// memory busy perc process lifetime   -1 = n/a
+        count_t	timems;		// milliseconds accounting   -1 = n/a
+        // value 0   for active process,
+        // value > 0 after termination
+        count_t	memnow;		// current    memory consumption in KiB
+        count_t	memcum;		// cumulative memory consumption in KiB
+        count_t	sample;		// number of samples
+    } gpu;
+};
+"""
+
+c_atop = cstruct()
+c_atop.load(atop_def)
+c_atop.load(atop_tstat_def, align=True)
+
+AtopRecord = TargetRecordDescriptor(
+    "linux/log/atop",
+    [
+        ("datetime", "ts"),
+        ("string", "process"),
+        ("string", "cmdline"),
+        ("varint", "tgid"),
+        ("varint", "pid"),
+        ("varint", "ppid"),
+        ("varint", "ruid"),
+        ("varint", "euid"),
+        ("varint", "suid"),
+        ("varint", "fsuid"),
+        ("varint", "rgid"),
+        ("varint", "egid"),
+        ("varint", "sgid"),
+        ("varint", "fsgid"),
+        ("varint", "nthr"),
+        ("boolean", "isproc"),
+        ("string", "state"),
+        ("varint", "excode"),
+        ("varint", "elaps"),
+        ("varint", "nthrslpi"),
+        ("varint", "nthrslpu"),
+        ("varint", "nthrrun"),
+        ("varint", "ctid"),
+        ("varint", "vpid"),
+        ("boolean", "wasinactive"),
+        ("string", "container"),
+        ("path", "filepath"),
+    ],
+)
+
+
+class AtopFile:
+    """Parse general task information of processes of an Atop log file.
+
+    Sources:
```

Comment:
Could you move the sources to the C-struct section please? That is the place where one usually expects to find them I guess...